docs/style.md

```diff
@@ -196,6 +196,53 @@ external crates. The surprise factor for derivations sharing a name with the sta
 library traits (`Display`) is reduced and it also acts as natural mechanism to tell apart names
 prone to collision (`Serialize`), all without needing to look up the list of imports.
 
+### Visibility
+
+"Minimize visibility" is a standard best practice, but rust has some extra
+specifics. In Rust, there are two granularities for visibility: visibility of
+items within a single crate, and visibility of items across the crates.
+Cross-crate APIs often form important architectural boundaries affecting the
+whole application, while what happens inside a crate concerns only the crate
+itself.
+
+The difference between `pub` (making items visible to other crates) and
+`pub(crate)` (making items visible within the crate) is big, while the
+difference between `pub(crate)` and private is relatively small. Be mindful when
+exposing APIs across crates with `pub`.
+
+For a top-level item, use `pub` only when it must be reachable from outside of
+the crate, don't rely on effective visibility:
+
+```rust
+// GOOD
+pub(crate) mod auth {
+    pub(crate) check_password(password: &str) -> Result<(), ()>
+}
+
+// BAD
+pub(crate) mod auth {
+    pub check_password(password: &str) -> Result<(), ()>
+}
+```
+
+**Rationale:** it's useful to tell at a glance that `pub` items form crate's
+public API without checking all the ancestor modules.
+
+For fields and methods, it's OK to use `pub` even if the type itself is not
+`pub`:
+
+```rust
+// GOOD
+pub(crate) struct Person {
+    pub first_name: String,
+    pub last_name: String,
+}
+
+impl Person {
+    pub fn full_name(&self) -> String;
```

Comment:
So in this example, can code outside of the crate access `full_name`?  Does that depend upon the visibility of `struct Person`?  If the visibility of `full_name` can change depending upon the visibility of `struct Person`, then doesn't the argument from above apply here?  `impl Person` can be very far away from `struct Person` and I will have to hunt down that definition to confirm what kind of visibility `full_name` has.

Reply:
I don’t understand that use-case to be honest.  If you think something shouldn’t be public, make it `pub(crate)`.  Why does it matter that a struct has a `pub` field?

Reply:
I am looking at it from the reader's point of view.  The rationale above says: 

```
**Rationale:** it's useful to tell at a glance that `pub` items form crate's public API without checking all the ancestor modules.
```

So the way I see it, if you have a pub method in a non-pub struct, that is breaking this rationale.

Reply:
Ah, yes, I see your point.  To be honest I don’t agree with that rationale either but I didn’t want to drag the discussion. ;)